x-pack/legacy/plugins/apm/public/components/app/TransactionDetails/WaterfallWithSummmary/WaterfallContainer/Waterfall/waterfall_helpers/waterfall_helpers.ts

```diff
@@ -119,6 +135,37 @@ function getSpanItem(span: Span): IWaterfallItemSpan {
   };
 }
 
+function getErrorItem(error: APMError): IWaterfallItemError {
+  return {
+    id: error.error.id,
+    parentId: error.parent?.id,
+    serviceName: error.service.name,
+    message: error.error.log?.message || error.error.exception?.[0]?.message,
+    timestamp: error.timestamp.us,
+    offset: 0,
+    skew: 0,
+    docType: 'error',
+    error,
+    duration: 0
+  };
+}
+
+function getAgentMarks(transaction?: Transaction): IWaterfallItemAgentMark[] {
+  const agent = transaction?.transaction.marks?.agent;
+  if (!agent) {
+    return [];
+  }
+  return Object.entries(agent).map(
+    ([name, ms]) =>
+      ({
+        id: name,
+        name,
```

Comment:
Having `name` looks a bit odd. Maybe another reason for keeping the custom data in a specific field:
```js
{
  custom: { customData... },
  offset: ms * 1000,
  docType: 'agentMark'
}
```